python/pyspark/sql/tests/test_udf.py

```diff
@@ -334,68 +334,118 @@ def test_udf_with_filter_function(self):
         self.assertEqual(sel.collect(), [Row(key=1, value="1")])
 
     def test_udf_with_variant_input(self):
-        df = self.spark.range(0, 10).selectExpr("parse_json(cast(id as string)) v")
-
-        u = udf(lambda u: str(u), StringType())
-        with self.assertRaises(AnalysisException) as ae:
-            df.select(u(col("v"))).collect()
-
-        self.check_error(
-            exception=ae.exception,
-            errorClass="DATATYPE_MISMATCH.UNSUPPORTED_UDF_INPUT_TYPE",
-            messageParameters={"sqlExpr": '"<lambda>(v)"', "dataType": "VARIANT"},
-        )
+        for arrow_enabled in ["false", "true"]:
+            with self.sql_conf({"spark.sql.execution.pythonUDF.arrow.enabled": arrow_enabled}):
```

Comment:
We don't need this here. There is `PythonUDFArrowTests` sharing tests in `BaseUDFTestsMixin` with the config enabled.